ttg/ttg/parsec/ttg.h

```diff
@@ -1272,6 +1274,9 @@ namespace ttg_parsec {
 
     bool m_defer_writer = TTG_PARSEC_DEFER_WRITER;
 
+    std::vector<ttg::meta::detail::constraint_callback_t<keyT>> constraints_check;
```

Comment:
I'll put some comments there. These are two different callbacks:

1) `constraints_check` hands the task over to the constraint, which will either tell us directly that it is not constrained (i.e., can be executed) or stores the key for later release through a callback (the listener action) and tells us that the key is indeed constrained.
2) `constraints_complete` tells the constraint that the task has executed (so that the constraint can check for new tasks to release through the listener action)